pkg/jobrunaggregator/jobruntestcaseanalyzer/analyzer.go

```diff
@@ -0,0 +1,278 @@
+package jobruntestcaseanalyzer
+
+import (
+	"context"
+	"encoding/xml"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorapi"
+	"github.com/openshift/ci-tools/pkg/jobrunaggregator/jobrunaggregatorlib"
+	"github.com/openshift/ci-tools/pkg/junit"
+)
+
+const (
+	installTest = "cluster install should succeed"
+)
+
+type requiredPasses struct {
+	testName               string
+	requiredNumberOfPasses int
+}
+
+// JobRunTestCaseAnalyzerOptions
+// 1. get a list of jobs from big query that meet the passed criteria: platform, network etc.
+// 2. finds jobruns for matching jobs for the specified payload tag
+// 3. reads all junit for the each jobrun
+// 4. constructs a synthentic junit that check minimum required succuessful passes for configured tests
+type JobRunTestCaseAnalyzerOptions struct {
+	payloadTag          string
+	workingDir          string
+	jobRunStartEstimate time.Time
+	clock               clock.Clock
+	timeout             time.Duration
+	platform            string
+	infrastructure      string
+	network             string
+	ciDataClient        jobrunaggregatorlib.CIDataClient
+	ciGCSClient         jobrunaggregatorlib.CIGCSClient
+	gcsClient           *storage.Client
+	requiredPasses      []requiredPasses
+	payloadInvocationID string
+	jobGCSPrefixes      *[]jobGcsPrefix
+}
+
+func (o *JobRunTestCaseAnalyzerOptions) getJobListFromNames(jobNames []string, allJobs []jobrunaggregatorapi.JobRow) ([]jobrunaggregatorapi.JobRow, error) {
+	jobMap := map[string]jobrunaggregatorapi.JobRow{}
+	for i := range allJobs {
+		job := allJobs[i]
+		jobMap[job.JobName] = job
+	}
+	jobs := []jobrunaggregatorapi.JobRow{}
+	for _, jobName := range jobNames {
+		if job, ok := jobMap[jobName]; !ok {
+			return jobs, fmt.Errorf("invalid job %s", jobName)
+		} else {
+			jobs = append(jobs, job)
+		}
+	}
+	return jobs, nil
+}
+
+func getJobInfrastructure(name string) string {
+	if strings.Contains(name, "upi") {
+		return "upi"
+	}
+	return "ipi"
```

Comment:
This is probably a nit, but I'd explicitly check if "ipi" occurs within `names` too and return "ipi" if true.

I realize there is only "ipi" and "upi" today so it doesn't matter; but if a new one is added, this code will return "ipi" for anything not "upi".

Reply:
Job names do not always have "upi" or "ipi". When upi is in the name, it is upi, otherwise, it is considered ipi.